cmd/root.go

```diff
@@ -84,36 +84,42 @@ func sendEmail(to, subject, body string) error {
 	return gomail.Send(s, m)
 }
 
-func sendEmails(matches []MatchPair) {
+func sendEmails(matches []MatchPair) ([]string, error) {
+	// TODO: add retry logic for failed emails (maybe use a queue)
+	// instead of sending each email individually,
+	// generate all the emails and send them in bulk to avoid individual errors
+	emailIssues := []string{}
 	subject := "Your Secret Santa Match!"
 
 	templateFile := "./templates/email_template.html"
 	tmpl, err := template.ParseFiles(templateFile)
 	if err != nil {
-		log.Fatalf("Error parsing email template: %v", err)
+		return emailIssues, fmt.Errorf("failed to parse email template: %w", err)
 	}
 
 	for _, match := range matches {
 		var emailBodyBuffer = &strings.Builder{}
 		err := tmpl.Execute(emailBodyBuffer, match)
 		if err != nil {
-			log.Printf("Error executing template for %s: %v", match.From.Email, err)
-			continue
+			emailIssues = append(emailIssues, fmt.Sprintf("failed to execute template for %s: %v", match.From.Email, err))
```

Comment:
If we do have an issue here, we actually still want to call `continue`, no point running the rest of the code in this iteration?